### PR TITLE
CS-125: add the ability to select a ring audio device

### PR DESCRIPTION
**Pull request by alexn29**
image

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -588,6 +604,16 @@
                 elementId: "media_device_custom",
               },
               by_name: {
+                ringoutputselector: {
+                  events: {
+                    onchange: (event) => {
+                      let element = event.srcElement;
```
Comment (mavriciomart) [inline index.html:610]: I think this can be `const`

File: src/lwpMediaDevices.js [hunk 2]
```diff
@@ -1309,13 +1396,32 @@ export default class extends lwpRenderer {
...
+  async _shimEnumerateDevices() {
+    const devices = await navigator.mediaDevices.enumerateDevices();
+    const outputDevices = devices.filter(
+      (device) => device.kind === "audiooutput"
+    );
+    let ringoutputDevices = [];
+
+    outputDevices.forEach((device) => {
+      ringoutputDevices.push(this._outputDeviceToRingDevice(device));
+    });
```
Comment (mavriciomart) [inline src/lwpMediaDevices.js:1413]: I think you can optimize this a little bit more by doing something like this:
```js
const devices = await navigator.mediaDevices.enumerateDevices();
const ringoutputDevices = [];

devices.forEach((device) => {
  if (device.kind !== "audiooutput") return;
  ringoutputDevices.push(this._outputDeviceToRingDevice(device));
});
```

Comment (alexn29) [reply src/lwpMediaDevices.js]: Done ✅

File: index.html (hunk 1 above)
Comment (alexn29) [reply index.html]: Done ✅

**Review by Irvandoval — APPROVED** [review]

**Review by Mariotutal — APPROVED** [review]

File: src/lwpMediaDevices.js [hunk 3]
```diff
@@ -731,6 +779,43 @@ export default class extends lwpRenderer {
 
   /** Helper functions */
 
+  async _changeRingOutputDevice(preferedDevice) {
+    if (this._config.ringoutput.mediaElement.element) {
+      return this._config.ringoutput.mediaElement.element
+        .setSinkId(preferedDevice.id)
+        .then(() => {
+          this._availableDevices["ringoutput"].forEach((availableDevice) => {
```
Comment (masmerino13) [inline src/lwpMediaDevices.js:787]: why not a `.map` to return the array with the updated values?

File: src/lwpMediaDevices.js [hunk 4]
```diff
@@ -402,7 +420,7 @@ export default class extends lwpRenderer {
         this._config[deviceKind].mediaElement.create &&
         this._config[deviceKind].enabled
       ) {
-        if (deviceKind === "audiooutput") {
+        if (deviceKind == "audiooutput" || deviceKind == "ringoutput") {
```
Comment (masmerino13) [inline src/lwpMediaDevices.js:423]:
```suggestion
        if (["audiooutput", "ringoutput"].includes(deviceKind)) {
```

**Review by masmerino13 — APPROVED** [review]
Looks good for me, I just leave some comments/suggestions,

File: src/lwpMediaDevices.js (hunk 3 above)
Comment (alexn29) [reply src/lwpMediaDevices.js:787]: I did it in that way following the same code guidelines of the project

File: src/lwpMediaDevices.js (hunk 4 above)
Comment (alexn29) [reply src/lwpMediaDevices.js]: Done

**Pull request merged**